website/blog/2025-03-19-20-must-try-mongodb-queries-operations-ferretdb.md

```diff
@@ -0,0 +1,436 @@
+
+Due to a price drop for "Pride and Prejudice" from the initial price of `$19.99` to `$17.99`, let's update the document to reflect the change.
+The `$set` operator is your command to use – it lets you modify specific fields without touching the rest of the document.
+
+```js
+db.books.updateOne(
+  { _id: 'pride_prejudice_1813' },
+  { $set: { 'price.value': 17.99 } }
+)
+```
+
+### 3. Sort documents
+
+Sorting is crucial for presenting data in a meaningful way.
+For instance, you might want to list books by publication date, price, or rating.
+
+Let's sort books by publication date in descending order:
+
+```js
+db.books.find().sort({ 'publication.date': -1 })
+```
+
+### 4. Count documents
+
+Are you curious about the number of books you have in the database?
+
+`countDocuments()` gives you a quick total of all documents – quite handy when dealing with large datasets or checking if an import was successful.
+
+```js
+db.books.countDocuments()
+```
+
+This should return `5` as the count for all the documents in the collection.
+
+### 5. Query nested array with `$elemMatch`
+
+Books often have multiple authors, genres, and availability formats.
+What if you only want books written by specific authors, e.g.
+British authors?
+`$elemMatch` helps filter specific values inside nested arrays.
+
+```js
+db.books.find({ authors: { $elemMatch: { nationality: 'British' } } })
+```
+
+### 6. Logical queries
+
+Need to find books that match multiple conditions?
+Logical operators let you chain queries, which is great for complex filtering.
+Let's find books that are either in the "Romance" genre or were published before 1900:
+
+```js
+db.books.find({
+  $or: [
+    { genres: 'Romance' },
+    { 'publication.date': { $lt: '1900-01-01T00:00:00Z' } }
+  ]
+})
+```
+
+### 7. Lookup (Join Queries)
+
+Sometimes, you need to combine data from multiple collections – just like SQL joins in relational databases.
+This is useful when you store related data in different collections but want to retrieve them together in a single query.
+
+Suppose there's another collection `publishers` with more details on the publishers, as shown below:
+
+```js
+db.publishers.insertMany([
+  {
+    _id: 1,
+    name: 'T. Egerton',
+    location: 'London, United Kingdom',
+    established: 1780
+  },
+  {
+    _id: 2,
+    name: 'Harper & Brothers',
+    location: 'New York, United States',
+    established: 1817
+  },
+  {
+    _id: 3,
+    name: 'Lackington, Hughes, Harding, Mavor & Jones',
+    location: 'London, United Kingdom',
+    established: 1790
+  }
+])
+```
+
+You can join the data from both collections (`books` and `publishers`) using `$lookup`.
+
+```js
+db.books.aggregate([
+  {
+    $lookup: {
+      from: 'publishers',
+      localField: 'publication.publisher.name',
+      foreignField: 'name',
+      as: 'publisher_details'
+    }
+  }
+])
+```
+
+This should fetch the books and their corresponding publishers' details in a single query.
+
+## Authentication
```

Comment:
If we have to describe that, let's move it to the last section so queries, aggregations, and indexes are closer to each other.